### PR TITLE
TRUNK-5815: Add junit 5 dependencies

### PR DESCRIPTION
**Pull request by achilep**
## Description of what I changed

 Adding the following dependencies on the main Pom file and the pom test file  :
org.junit.jupiter:junit-jupiter-api:5.6.2
org.junit.jupiter:junit-jupiter-engine:5.6.2
org.junit.vintage:junit-vintage-engine:5.6.2

## Issue I worked on

see https://issues.openmrs.org/browse/TRUNK-5815
## Checklist: I completed these to help reviewers :)

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [ ] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

**Comment by HerbertYiga** [comment]
@teleivo do you think we shall execute any tests in a junit 4 environment that might require us adding junit-platform-runner dependecy?

**Comment by teleivo** [comment]
@HerbertYiga we will use the vintage engine which will allow us to still run Junit 4 tests while slowly migrating to Junit 5. For details see https://issues.openmrs.org/browse/TRUNK-5811

File: pom.xml [hunk 1]
```diff
@@ -469,11 +469,31 @@
...
+			</dependency>
 			<dependency>
 				<groupId>org.mockito</groupId>
 				<artifactId>mockito-core</artifactId>
 				<version>3.3.3</version>
 			</dependency>
+			<dependency>
+				<groupId>org.mockito</groupId>
+				<artifactId>mockito-junit-jupiter</artifactId>
+				<version>3.3.3</version>
```
Comment (teleivo) [inline pom.xml:495]: you can also extract the `mockitoVersion` into the properties next to the `junitVersion` now that we depend on 2 mockito packages that should stay in sync

Comment (achilep) [reply pom.xml]: @teleivo  corrected

**Pull request closed without merging**